server/controllers/productDetails.js

```diff
@@ -0,0 +1,12 @@
+const { getProductDetailsQuery } = require('../database/queries/products');
+
+const productDetails = (req, res, next) => {
+  const { id } = req.body;
```

Comment:
maybe it'll be better if id as part of the endpoint, handled by params instead of the request body.